### PR TITLE
feat: improve `vercel integration add <query_or_slug>`

### PR DESCRIPTION
**Pull request by atinux**
Before:

```bash
$ vercel integration add postgres
Error: Failed to get integration "postgres": Not found (404)
```

After:

```bash
$ vercel integration add postgres
? Found 6 integrations matching "postgres". Pick one to install: (Use arrow keys)
❯ Amazon Aurora PostgreSQL (aws/aws-apg) - Full PostgreSQL compatibility
  Amazon Aurora DSQL (aws/aws-dsql) - Serverless PostgreSQL
  Prisma Postgres (prisma/prisma-postgres) - Instant Serverless Postgres
  Nile (nile) - Postgres re-engineered for B2B
  Supabase (supabase) - Postgres backend
  Neon (neon) - Serverless Postgres
```

Adding integration with slug works like before, when we hit a 404, we fallback to the `vercel integration discovery <query>` and see if it has results, if so, we suggest it to the end user instead, improve the DX.

If only one integration is available, it ask to install it:

```bash
$ vercel integration add sqlite
? Install Turso (tursocloud/database)? (Y/n)
```

### What about agents?

If terminal is not TTY, we throw an error with the list of options so the agent can ask the user what postgres database it prefers to install:

CleanShot 2026-03-31 at 17 01 57@2x

### CODE REVIEW

File: packages/cli/src/util/integration/fetch-integration.ts [hunk 1]
```diff
@@ -1,37 +1,174 @@
...
+
+  if (matches.length === 1) {
+    const match = matches[0];
+    if (client.stdin.isTTY === true) {
+      const confirmed = await client.input.confirm(
+        `Install ${chalk.bold(match.name)} (${match.slug})?`,
+        true
+      );
+      if (!confirmed) {
+        return null;
```
Comment (tonypan2) [inline packages/cli/src/util/integration/fetch-integration.ts:133]: This `return null` path (user declines single-match confirmation) doesn't call `telemetry.trackCliArgumentIntegration()`. The old `fetchIntegrationWithTelemetry` used a `finally` block to guarantee tracking on every exit. All other exit paths in this new function track telemetry — this one is the gap.

*🤖 Written by Claude Code on behalf of @tonypan2*

File: packages/cli/src/commands/integration/add-auto-provision.ts [hunk 2]
```diff
@@ -76,19 +76,22 @@ export async function addAutoProvision(
...
+  const integration = await resolveAndFetchIntegration(
     client,
-    integrationSlug,
+    integrationSlugOrQuery,
     telemetry
   );
   if (!integration) {
     return 1;
   }
+  if (integration.productSlug) {
```
Comment (tonypan2) [inline packages/cli/src/commands/integration/add-auto-provision.ts]: Nit: this unconditionally overwrites `options.productSlug`. The equivalent code in `add.ts` has a `!productSlug` guard:

```ts
if (integration.productSlug && !productSlug) {
  productSlug = integration.productSlug;
}
```

Should this also check `!options.productSlug` for consistency?

*🤖 Written by Claude Code on behalf of @tonypan2*

File: packages/cli/test/unit/commands/integration/add-auto-provision.test.ts [hunk 3]
```diff
@@ -2688,4 +2691,66 @@ describe('integration add (auto-provision)', () => {
       );
     });
   });
+
```
Comment (tonypan2) [inline packages/cli/test/unit/commands/integration/add-auto-provision.test.ts:2694]: Nice test coverage for the discover fallback! One edge case that might be worth adding: selecting a multi-product integration from discover (e.g., searching for something that resolves to `acme-multi/acme-db`). This would exercise the compound slug parsing at the bottom of `resolveAndFetchIntegration` and the `productSlug` propagation back to the caller. The mock data already has `acme-multi` with two products set up for this.

*🤖 Written by Claude Code on behalf of @tonypan2*

File: packages/cli/src/util/integration/fetch-integration.ts [hunk 4]
```diff
@@ -1,37 +1,174 @@
...
+    directError = error as Error;
+  }
+
+  output.spinner('Searching marketplace integrations...', 500);
+
+  let entries: DiscoverEntry[];
+  try {
+    const integrations = await fetchMarketplaceIntegrationsList(client);
+    entries = toDiscoverEntries(integrations);
+  } catch (_discoverError) {
```
Comment (tonypan2) [inline packages/cli/src/util/integration/fetch-integration.ts]: Nit: `_discoverError` is prefixed with underscore (convention for "intentionally unused") but it's actually used in the error message on line 106. Should this be `discoverError` instead?

*🤖 Written by Claude Code on behalf of @tonypan2*

**Review by tonypan2 — APPROVED** [review]

**Comment by tonypan2** [comment]
11 AI-generated findings triaged, 4 posted after human review.
AI review: 22m 53s | Human triage: 56m 51s | Total: 79m 44s

*🤖 Written by Claude Code on behalf of @tonypan2*

**Comment by atinux** [comment]
Thank you for the review @tonypan2, should be good now!

**Comment by atinux** [comment]
Alright all tests pass, should I merge myself or you'll do it @tonypan2 ?

**Comment by tonypan2** [comment]
@atinux merged! (you are free to merge too if you see the big green button :))
